substrate/frame/revive/src/precompiles/builtin/system.rs

```diff
@@ -0,0 +1,68 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{
+	precompiles::{BuiltinAddressMatcher, BuiltinPrecompile, Error, Ext},
+	vm::RuntimeCosts,
+	Config,
+};
+use alloc::vec::Vec;
+use alloy_core::sol;
+use core::{marker::PhantomData, num::NonZero};
+use sp_core::hexdisplay::AsBytesRef;
+
+pub struct System<T>(PhantomData<T>);
+
+sol! {
+	interface ISystem {
+		/// Computes the BLAKE2 256-bit hash on the given input.
+		function hash_blake2_256(bytes memory input) external pure returns (bytes32 digest);
+	}
+}
+
+impl<T: Config> BuiltinPrecompile for System<T> {
+	type T = T;
+	type Interface = ISystem::ISystemCalls;
+	const MATCHER: BuiltinAddressMatcher = BuiltinAddressMatcher::Fixed(NonZero::new(901).unwrap());
```

Comment:
I've used `901` now, but let me know if you have another number range in mind. This is loosely inspired from how Moonbeam does it (https://docs.moonbeam.network/builders/ethereum/precompiles/overview).